### PR TITLE
Refactor pause() to skip refreshing quota

### PR DESCRIPTION
**Pull request by xuebinsu**
Currently, diskquota.pause() only takes effect on quota checking.
Bgworkers still go over the loop to refreshing quota even if diskquota
is paused. This wastes computation resources and can cause flaky issues.

This fix makes bgworkers skip refreshing quota when the user pauses
diskquota entirely to avoid those issues. Table sizes can be updated
correctly after resume.

### CODE REVIEW

File: diskquota.c [hunk 1]
```diff
@@ -79,6 +79,16 @@ static int	num_db = 0;
  */
 bool *diskquota_paused = NULL;
 
+static inline bool
+is_worker_paused()
+{
+
```
Comment (higuoxing) [inline diskquota.c:85]: Nit: remove this blank line.
```suggestion
```

**Review by higuoxing — APPROVED** [review]
LG!

File: tests/regress/diskquota_schedule [hunk 2]
```diff
@@ -4,6 +4,7 @@ test: test_worker_epoch
 test: test_relation_size
 test: test_relation_cache
 test: test_uncommitted_table_size
+test: test_pause_and_resume
```
Comment (beeender) [inline tests/regress/diskquota_schedule:7]: any special reason we move this case here?

File: diskquota.c [hunk 3]
```diff
@@ -79,6 +79,16 @@ static int	num_db = 0;
...
+	LWLockAcquire(diskquota_locks.paused_lock, LW_SHARED);
```
Comment (beeender) [inline diskquota.c:86]: We can simply remove the `pause_lock` by using pg's atomic operations.
https://github.com/greenplum-db/gpdb/blob/master/src/include/port/atomics.h

It is not related to this PR. If needed, we can do in another PR.

File: tests/regress/diskquota_schedule (hunk 2 above)
Comment (xuebinsu) [reply tests/regress/diskquota_schedule:7]: Because `pause()` and `resume()` are fundamental functions and need to be used in other tests. 

For example, we need to `SELECT diskquota.pause();` before `DROP EXTENSION diskquota;` to avoid possible deadlocks in `test_extension`.

File: diskquota.c (hunk 3 above)
Comment (xuebinsu) [reply diskquota.c:85]: Good idea. In fact, we have some other shared-memory variables that needs to be refactored that way such as `diskquota_hardlimit`. Let's leave it to the next PR.

File: tests/regress/diskquota_schedule (hunk 2 above)
Comment (higuoxing) [reply tests/regress/diskquota_schedule:7]: I think whether `pause()` or not shouldn't affect dropping the extension. Not sure if we can fix it in https://github.com/greenplum-db/diskquota/blob/d04e5ef1bebe11534bed763786e49ea4bb4d686c/diskquota_utility.c#L449 by pausing the extension automatically before dropping it.

Comment (xuebinsu) [reply tests/regress/diskquota_schedule:7]: @higuoxing Good idea. will check the hook later.

**Review by beeender — APPROVED** [review]
:+1:
